apps/examples/timer/timer_main.c

```diff
@@ -90,57 +90,40 @@
 #define TIMER_THEAD_SIZE                1024
 #define TIMER_THEAD_PRIORITY            100
 
-/****************************************************************************
- * Private Data
- ****************************************************************************/
-static sem_t g_sem;
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
-static void timer_sighandler(int signo, FAR siginfo_t *siginfo,
-							 FAR void *context)
-{
-	/*
-	 * Does nothing in this example except for increment a count of signals
-	 * received.
-	 *
-	 * NOTE: The use of signal handler is not recommended if you are concerned
-	 * about the signal latency.  Instead, a dedicated, high-priority thread
-	 * that waits on sigwaitinfo() is recommended.  High priority is required
-	 * if you want a deterministic wake-up time when the signal occurs.
-	 */
-	sem_post(&g_sem);
-}
 
-static pthread_addr_t timer_thread(pthread_addr_t arg)
+static pthread_addr_t hrt_thread(pthread_addr_t arg)
 {
-	int nbr = 0;
-	int i = (int)arg;
+	int count = (int)arg;
 
-	while (i--) {
-		sem_wait(&g_sem);
-		fprintf(stdout, "time limits(%d)\n", ++nbr);
-	}
+	sigset_t sig_set;
+	sigemptyset(&sig_set);
+	sigaddset(&sig_set, EXAMPLE_TIMER_SIGNO);
+
+	pthread_sigmask(SIG_BLOCK, &sig_set, NULL);
 
+	while (count--) {
+		fprintf(stdout, "Waiting on a signal...\n");
+		if (sigwaitinfo(&sig_set, NULL) == EXAMPLE_TIMER_SIGNO)
+			fprintf(stdout, "signal received\n");
+	}
+	fprintf(stdout, "signal received:%d\n", (int)arg);
+	pthread_sigmask(SIG_UNBLOCK, &sig_set, NULL);
 	pthread_exit(NULL);
 
 	return NULL;
 }
 
-static pthread_t create_timer_thread(void *arg)
+static pthread_t create_hrt_thread(void *arg)
```

Comment:
Keep same earlier name, i dont think we should use hrt here.

Reply:
Done